tests/test_incident.py

```diff
@@ -117,3 +126,89 @@ def test_inc_revisions(mock_good):
     assert incident.revisions_with_fallback("x86_64", "12-SP5")
     assert not incident.revisions_with_fallback("aarch64", "12")
     assert not incident.revisions_with_fallback("aarch64", "12-SP5")
+
+
+class MockResponse:
+    # TODO: collect all instances where the same pattern is used and refactor,
+    # see openSUSE/qem-bot/issues/161
+    def __init__(self, url, json_data, extra_data=None):
+        self.url = url  # the url helps us mock different responses
+        self.json_data = json_data
+        self.extra_data = extra_data
+
+    def mock_comments(self, job=1777, incident=24618):
+        return [
+            {
+                "bugrefs": [],
+                "created": "2024-01-30 16:04:56 +0000",
+                "id": job,
+                "renderedMarkdown": None,
+                "text": "label:linked Job mentioned in https://progress.opensuse.org/issues/154156",
+                "text": "@review:acceptable_for:incident_%s:openqa#1337" % incident,
+                "updated": "2024-01-30 16:04:56 +0000",
+                "userName": "system",
+            }
+        ]
+
+    def json(self):
+        if "openqa" in self.url:
+            self.json_data = []
+            if "1777" in self.url:
+                self.json_data = self.mock_comments()
+        if "qam" in self.url:
+            pass  # right now we don't need to mock anything else for requests to the dashboard
+        # leave comment for future debugging purposes
+        # logger.debug("Mocking json: %s", self.json_data)
+        return self.json_data
+
+    def __repr__(self):
+        return f"<MockResponse for {self.url}>"
+
+
+def mock_get(url, extra_data=None, headers=None):
+    return MockResponse(
+        url=url,
+        json_data=[
+            {"status": "passed", "job_id": 1},
+            {"status": "failed", "job_id": 1777},  # Accept the turk
+            {
+                "status": "softfailed",
```

Comment:
@kraih this looks like it's mocking a response from qem-dashboard that can actually never happen in reality as the dashboard would not return a status "softfailed", am I right?

Reply:
I also guess so. The status from dashboard should be seen from the perspective of [approval job](https://github.com/openSUSE/qem-bot/pull/154/files#diff-d44cab80bf43c62865043529f6e9883cc12ab37d49f5fb5af9e1bdc7c012dcaaL168) which only has binary result: either approve or not. An openQA softfail should not block approval and hence is masked by dashboard as passed (as already linked in another conversation).

Reply:
This was a leftover